### PR TITLE
[Cost-2031] - Adding savingsplan to postgres

### PR DESCRIPTION
**Pull request by shannon-donahue**
#### Testing Instructions:
1. Pull and Checkout branch
2. Bring up Docker 
3. Create and Load test costumer Data
4. hit the following Endpoints:
  For the following tests the values for blended and unblended should match as they do in the nise data. Savings plan should be the only cost type which differs in value.
```
Endpoint for unblended_cost:
http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?filter%5Btime_scope_value%5D=-1&check_tags=True

Endpoint for blended_cost:
http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?filter%5Btime_scope_value%5D=-1&check_tags=True&cost_type=blended_cost

Endpoint for savingsplan:
http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?filter%5Btime_scope_value%5D=-1&check_tags=True&cost_type=savingsplan_effective_cost

```
Repeat steps 2-4 for both bringing docker up in Trino and postgres:

### CODE REVIEW

**Review by maskarb — CHANGES_REQUESTED** [review]
Looks good so far! Just a couple comments/suggestions:

File: koku/reporting/provider/aws/models.py [hunk 1]
```diff
@@ -69,7 +69,7 @@
     "lineItem/UnblendedCost",
     "lineItem/BlendedRate",
     "lineItem/BlendedCost",
-    "pricing/publicOnDemandCost",
+    "savingsPlan/SavingsPlanEffectiveCost" "pricing/publicOnDemandCost",
```
[the remark this answers is not in the document]
Comment (shannon-donahue) [reply koku/reporting/provider/aws/models.py]: Good catch must have been a typo on one of the edits made, should be pushed up and fixed now!

File: koku/reporting/provider/aws/sql/views/_20211109/reporting_aws_compute_summary_20211109.sql [hunk 2]
```diff
@@ -0,0 +1,59 @@
+DROP INDEX IF EXISTS aws_compute_summary;
```
[the remark this answers is not in the document]
Comment (shannon-donahue) [reply koku/reporting/provider/aws/sql/views/_20211109/reporting_aws_compute_summary_20211109.sql]: This whole folder should have been deleted must have missed the last file should be taken care of now

File: koku/masu/database/aws_report_db_accessor.py [hunk 3]
```diff
@@ -432,16 +432,26 @@ def populate_markup_cost(self, provider_uuid, markup, start_date, end_date, bill
...
-                    )
+                MARKUP_MODELS_BILL_AWS.objects.filter(cost_entry_bill_id=bill_id, **date_filters).update(
+                    markup_cost=(F("unblended_cost") * markup)
+                )
+                MARKUP_MODELS_BILL_AWS.objects.filter(cost_entry_bill_id=bill_id, **date_filters).update(
+                    markup_cost_blended=(F("blended_cost") * markup)
+                )
+                MARKUP_MODELS_BILL_AWS.objects.filter(cost_entry_bill_id=bill_id, **date_filters).update(
+                    markup_cost_savingsplan=(F("savingsplan_effective_cost") * markup)
+                )
```
[the remark this answers is not in the document]
Comment (shannon-donahue) [reply koku/masu/database/aws_report_db_accessor.py]: I tested this out real quick I wasn't sure if you could do more than one update at once but it seems like its all working, so definitely cleaned up the code a lot thanks for the suggestion!

File: koku/reporting/migrations/0207_savingsplan_postgres.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
+# Generated by Django 3.1.13 on 2021-11-15 16:29
```
Comment (Red-HAP) [inline koku/reporting/migrations/0207_savingsplan_postgres.py:1]: Please break this out into its own PR. That way, we can go ahead and apply it to the DB and have these new fields in place.

File: koku/masu/database/sql/aws/reporting_aws_compute_summary_by_service_p.sql [hunk 5]
```diff
@@ -59,6 +63,8 @@ INSERT INTO {{schema | sqlsafe}}.reporting_aws_compute_summary_by_service_p (
             SUM(blended_cost) AS blended_cost,
             SUM(savingsplan_effective_cost) AS savingsplan_effective_cost,
             SUM(markup_cost) AS markup_cost,
+            SUM(markup_cost_blended) AS markup_cost_blended,
+            SUM(markup_cost_savingsplan) AS markup_cost_savingsplan,
```
Comment (Red-HAP) [inline koku/masu/database/sql/aws/reporting_aws_compute_summary_by_service_p.sql:67]: For all instances of `markup_cost_blended` and `markup_cost_savingsplan` and `savingsplan_effective_cost` in the SQL files:
```suggestion
            SUM(coalesce(savingsplan_effective_cost, 0.0::numeric(33,15))) AS savingsplan_effective_cost,
            SUM(markup_cost) AS markup_cost,
            SUM(coalesce(markup_cost_blended, 0.0::numeric(33,15))) AS markup_cost_blended,
            SUM(coalesce(markup_cost_savingsplan, 0.0::numeric(33,15))) AS markup_cost_savingsplan,
```

File: koku/masu/database/sql/reporting_awscostentrylineitem_daily.sql [hunk 6]
```diff
@@ -23,6 +23,7 @@ CREATE TEMPORARY TABLE reporting_awscostentrylineitem_daily_{{uuid | sqlsafe}} A
         sum(li.unblended_cost) as unblended_cost,
         max(li.blended_rate) as blended_rate,
         sum(li.blended_cost) as blended_cost,
+        sum(li.savingsplan_effective_cost) as savingsplan_effective_cost,
```
Comment (Red-HAP) [inline koku/masu/database/sql/reporting_awscostentrylineitem_daily.sql:26]: Unless null values are OK, I suggest this:
```suggestion
        sum(coalesce(li.savingsplan_effective_cost, 0.0::numeric(33,15))) as savingsplan_effective_cost,
```

**Review by Red-HAP — CHANGES_REQUESTED** [review]

File: koku/reporting/migrations/0207_savingsplan_postgres.py (hunk 4 above)
Comment (abaiken) [reply koku/reporting/migrations/0207_savingsplan_postgres.py]: Broke this into https://github.com/project-koku/koku/pull/3291 - once this is merged, we will update this PR.

File: koku/masu/database/sql/aws/reporting_aws_compute_summary_by_service_p.sql (hunk 5 above)
Comment (abaiken) [reply koku/masu/database/sql/aws/reporting_aws_compute_summary_by_service_p.sql]: Updated for `markup_cost_blended`, `markup_cost_savingsplan`, and `savingsplan_effective_cost` 👍

File: koku/masu/database/sql/reporting_awscostentrylineitem_daily.sql (hunk 6 above)
Comment (abaiken) [reply koku/masu/database/sql/reporting_awscostentrylineitem_daily.sql]: updated 👍

**Review by maskarb — APPROVED** [review]
lgmt!

**Review by Red-HAP — APPROVED** [review]
